compiler/src/dmd/traits.d

```diff
@@ -972,9 +972,18 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
 
         if (e.ident == Id.hasMember)
         {
-            /* Take any errors as meaning it wasn't found
-             */
-            ex = ex.trySemantic(scx);
+            bool ungag;
+            // do not gag the sema of the parent, as the member is not requested yet
+            auto die = ex.isDotIdExp();
+            die.e1 = expressionSemantic(die.e1, scx.push());
```

Comment:
This could result in a "Segmentation Fault" if `ex` is not a `DotIdExp`. Even if currently `ex` is a DotIdExp on all code paths it's possible that it could be modified in the future. At the very least I would add an assert.